frontend/lib/Repository/EndpointRepository/EndpointGateway.dart

```diff
@@ -34,7 +37,8 @@ class EndpointGateway {
   Future<EndpointData> updateCacheAndGetEndpointData(
       int id, int limit, int offset) {
     final Future<EndpointData> endpointDataFuture =
-        restRepository.getEndpointData(id, limit, offset);
+        restRepository.getEndpointData(
+            id, limit, offset, userGateway.user.tokenResponse.accessToken);
```

Comment:
Trochę nie podoba mi się wywołanie `userGateway.user.tokenResponse.accessToken`. Najlepiej by było, jakbyśmy mieli jakiegoś `Provider'a`, który dostarcza nam ten `accessToken`. Może to być klasa z metodą `getAccessToken`, a w środku mamy wywołanie `secureStorage.read(key: accessKey)`. Taką klasę można potem spokojnie wstrzykiwać w różnych `gateway'ach`